### PR TITLE
feat(user-settings): Search for user avatar also in the entities

### LINKED ISSUE #26838: 🚀 Feature: Support using Backstage User entity profile data for UserSettingsAvatar
### 🔖 Feature description

The user settings plugin should show profile data, in my case the profile picture, if it is defined by the Backstage User entity but not by the identity provider

### 🎤 Context

Right now only the Identity Provider data is considered, but not the Catalog data. 

This leads in my case in which the Catalog has the picture data and the IdentityProvider not to inconsistent UX. 
On the settings page the picture is empty and on the User entity page the profile picture is correctly shown.  

### ✌️ Possible Implementation

https://github.com/backstage/backstage/blob/17ecea82239f25a4ee89f9c59fe43e735d1c80a2/plugins/user-settings/src/components/useUserProfileInfo.ts#L31-L36

Should also call the CatalogAPI to gather optionally infos about the Catalog entity based on the return value of `getBackstageEntity`. 

### 👀 Have you spent some time to check if this feature request has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by deejay1**
## Hey, I just made a Pull Request!

If the identity system doesn't provide an avatar, but the existing user entity does have the correct annotation use that as the profile avatar.

Fixes #26838

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] ~Added or updated documentation~
- [x] Tests for new functionality and regression tests for bug fixes
- [x] ~Screenshots attached (for UI changes)~
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by deejay1** [comment]
Argh, seems like other things break with these changes, will take a look later, going back to draft

File: plugins/user-settings/src/components/useUserProfileInfo.ts [hunk 1]
```diff
@@ -22,16 +22,34 @@ import {
...
+    const identityProfile = await identityApi.getProfileInfo();
+    const backStageIdentity = await identityApi.getBackstageIdentity();
+    const catalogProfile = await catalogApi.getEntityByRef(
+      backStageIdentity.userEntityRef,
+    );
+    if (
+      identityProfile.picture === undefined &&
+      catalogProfile?.metadata?.annotations?.[
+        'backstage.io/user-profile-picture'
+      ]
```
Comment (secustor) [inline plugins/user-settings/src/components/useUserProfileInfo.ts:43]:
```suggestion
     isUserEntity(catalogProfile) && catalogProfile.spec.profile?.picture
```
I have meant the picture in the entity spec.
https://backstage.io/docs/features/software-catalog/descriptor-format#kind-user

File: plugins/user-settings/src/components/useUserProfileInfo.ts [hunk 2]
```diff
@@ -22,16 +22,34 @@ import {
...
+    ) {
+      identityProfile.picture =
+        catalogProfile.metadata.annotations[
+          'backstage.io/user-profile-picture'
+        ];
```
Comment (secustor) [inline plugins/user-settings/src/components/useUserProfileInfo.ts:48]:
```suggestion
      identityProfile.picture = catalogProfile.spec.profile.picture
```

File: plugins/user-settings/src/components/useUserProfileInfo.ts (hunk 1 above)
Comment (deejay1) [reply plugins/user-settings/src/components/useUserProfileInfo.ts:43]: oops ;)

Comment (deejay1) [reply plugins/user-settings/src/components/useUserProfileInfo.ts]: fixed

File: .changeset/silly-geckos-learn.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-user-settings': minor
+---
+
+Search for profile image also in User Entities
```
Comment (vinzscam) [inline .changeset/silly-geckos-learn.md:5]:
```suggestion
'@backstage/plugin-user-settings': patch
---

`useUserProfile` will now use the user's picture stored in the catalog as a fallback if the identity provider doesn't return a picture.
```

**Review by vinzscam — CHANGES_REQUESTED** [review]

**Review by vinzscam — APPROVED** [review]
🚀

**Review by secustor — APPROVED** [review]

File: plugins/user-settings/src/components/useUserProfileInfo.ts [hunk 4]
```diff
@@ -22,16 +22,30 @@ import {
...
+    const identityProfile = await identityApi.getProfileInfo();
+    const backStageIdentity = await identityApi.getBackstageIdentity();
+    const catalogProfile = (await catalogApi.getEntityByRef(
+      backStageIdentity.userEntityRef,
+    )) as unknown as UserEntity;
+    if (
+      identityProfile.picture === undefined &&
+      catalogProfile?.spec?.profile?.picture
+    ) {
+      identityProfile.picture = catalogProfile.spec.profile.picture;
```
Comment (freben) [inline plugins/user-settings/src/components/useUserProfileInfo.ts:44]: I don't know that it's safe to assume that you can actually mutate the output of the getProfileInfo call. Could you instead do `identityProfile = { ...identityProfile, picture: catalogProfile.spec.profile.picture }`?

**Review by freben — COMMENTED** [review]
OK just one small tweak

**Review by adowsky — APPROVED** [review]

**Comment by deejay1** [comment]
@freben hmm, the node 18 ci failed again :/

**Comment by deejay1** [comment]
@vinzscam @freben ok, build passed this time. Ready to merge

**Review by freben — APPROVED** [review]
Aight

**Pull request merged**